src/test/java/com/google/step/servlets/PromoCodeServletTest.java

```diff
@@ -1,12 +1,119 @@
 package com.google.step.servlets;
 
-import org.junit.Assert;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.mock;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Optional;
+
+import com.google.step.youtube.YoutubeInfoScraper;
+import com.google.step.youtube.PromoCode;
+
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
 
-// @RunWith(JUnit4.class)
+@RunWith(MockitoJUnitRunner.class)
 public final class PromoCodeServletTest {
-    // TODO(margaret): Implementation coming
-}
 
+    private PromoCodeServlet servlet;
+
+    private static final String CHANNEL_ID_NONEXISTENT = "CHANNEL_ID_NONEXISTENT";
+    private static final String CHANNEL_ID = "CHANNEL_ID";
+    private static final String UPLOAD_ID = "UPLOAD_ID";
+    private static final String VIDEO_ID = "VIDEO_ID";
+    private static final String IOEXCEPTION = "IOEXCEPTION";
+    private static final long MOCK_DATE = 0L;
```

Comment:
instead of MOCK_DATE being a long, make it a Date so it's typed:

private static final Date MOCK_DATE = new Date(0);